src/Serialization/ConstructingPayloadSerializer.php

```diff
@@ -4,13 +4,14 @@
 
 namespace EventSauce\EventSourcing\Serialization;
 
+use function assert;
+
 class ConstructingPayloadSerializer implements PayloadSerializer
 {
-    /**
-     * @param SerializablePayload $event
-     */
     public function serializePayload(object $event): array
     {
+        assert($event instanceof SerializablePayload);
```

Comment:
Overall, the assertion is a bit worse than the non-LSP-compliant type declaration here.

Would be better to have this declared, rather than asserted upon at runtime.

Reply:
Here I thought the only time this would occur if the developer incorrectly creates an event, for which this should only be triggered during tests. This is why the default test setup serialises the events upon storing them in the in-memory store. By using assert, this call can be effectively removed in production.